### PR TITLE
SDI-2115:issue334: Remove metric.String()

### PR DESCRIPTION
**Pull request by candysmurf**
### Summary of Changes
* removed metric.String() method
* updated relevant tests that use that method.
* provided the private toString function in each package.

### Tests done
* small and medium tests.

### CODE REVIEW

File: v1/plugin/collector_proxy_test.go [hunk 1]
```diff
@@ -46,7 +47,7 @@ func TestGetMetricTypes(t *testing.T) {
 			for _, m := range r.GetMetrics() {
 				tm := fromProtoMetric(m)
 				idx := fmt.Sprintf("%s.%d", tm.Namespace, tm.Version)
-				So(tm.Namespace.String(), ShouldEqual, metricMap[idx].Namespace.String())
+				So(toString(tm.Namespace.Strings()), ShouldEqual, toString(metricMap[idx].Namespace.Strings()))
```
Comment (kindermoumoute) [inline v1/plugin/collector_proxy_test.go]: Those test should test if 2 slices of string are equal. 
```
So(slice1, ShouldResemble, slice2)
```

File: v1/plugin/collector_proxy_test.go [hunk 2]
```diff
@@ -92,7 +93,7 @@ func TestCollectMetrics(t *testing.T) {
 			for _, v := range reply.GetMetrics() {
 				m := fromProtoMetric(v)
 				So(v.Tags, ShouldEqual, m.Tags)
-				So("/"+v.Namespace[0].Value, ShouldEqual, m.Namespace.String())
+				So("/"+v.Namespace[0].Value, ShouldEqual, toString(m.Namespace.Strings()))
```
Comment (kindermoumoute) [inline v1/plugin/collector_proxy_test.go]: Same here

File: v1/plugin/collector_proxy_test.go (hunk 1 above)
Comment (candysmurf) [inline v1/plugin/collector_proxy_test.go]: I cannot compare that as it'll never pass. The config is gone after conversion. That's expected from the prior discussion with Cody.

```
* /Users/egu/.gvm/pkgsets/go1.7/snap-1.7/src/github.com/intelsdi-x/snap-plugin-lib-go/v1/plugin/collector_proxy_test.go
  Line 51:
  Expected: '"plugin.Metric{Namespace:plugin.namespace{plugin.namespaceElement{Value:\"a\", Description:\"\", Name:\"\"}, plugin.namespaceElement{Value:\"b\", Description:\"\", Name:\"\"}, plugin.namespaceElement{Value:\"c\", Description:\"\", Name:\"\"}}, Version:2, Config:plugin.Config{\"key\":123}, Data:2, Tags:map[string]string{\"tag2\":\"value2\"}, Timestamp:time.Time{sec:63612521385, nsec:588941258, loc:(*time.Location){name:\"\", zone:[]time.zone(nil), tx:[]time.zoneTrans(nil), cacheStart:0, cacheEnd:0, cacheZone:(*time.zone)(nil)}}, Unit:\"int\", Description:\"\", lastAdvertisedTime:time.Time{sec:0, nsec:0, loc:(*time.Location)(nil)}}"'
  Actual:   '"plugin.Metric{Namespace:plugin.namespace{plugin.namespaceElement{Value:\"a\", Description:\"\", Name:\"\"}, plugin.namespaceElement{Value:\"b\", Description:\"\", Name:\"\"}, plugin.namespaceElement{Value:\"c\", Description:\"\", Name:\"\"}}, Version:2, Config:plugin.Config{}, Data:2, Tags:map[string]string{\"tag2\":\"value2\"}, Timestamp:time.Time{sec:63612521385, nsec:588941258, loc:(*time.Location){name:\"\", zone:[]time.zone(nil), tx:[]time.zoneTrans(nil), cacheStart:0, cacheEnd:0, cacheZone:(*time.zone)(nil)}}, Unit:\"int\", Description:\"\", lastAdvertisedTime:time.Time{sec:63612521385, nsec:589213540, loc:(*time.Location){name:\"\", zone:[]time.zone(nil), tx:[]time.zoneTrans(nil), cacheStart:0, cacheEnd:0, cacheZone:(*time.zone)(nil)}}}"'
  (Should resemble)!
```

File: v1/plugin/collector_proxy_test.go (hunk 2 above)
Comment (candysmurf) [inline v1/plugin/collector_proxy_test.go]: Same reason that I cannot compare them.

**Comment by candysmurf** [comment]
@kindermoumoute, good catch for /snap-plugin-collector-mock2-grpc/mock/mock_medium_test.go. I'll change that and update you.

File: v1/plugin/collector_proxy_test.go (hunk 1 above)
Comment (candysmurf) [inline v1/plugin/collector_proxy_test.go]: Security violation that's why configs dropped.

**Comment by candysmurf** [comment]
@IRCody & @kindermoumoute, if not comparing strings but array of strings, it requires comparing of each array element or  reflect.DeepEqual(). I'll have to add an additional function called Equal to swap out the toString([]string). Please let me know you want this.

**Comment by kindermoumoute** [comment]
LGTM

**Pull request merged**